cloudify_agent/api/factory.py

```diff
@@ -104,7 +104,8 @@ def new(self, logger=None, **attributes):
         :rtype: cloudify_agent.api.pm.base.Daemon
         """
         name = attributes.get('name')
-        if name:
+        is_agents_transfer = attributes.get('agents_transfer_mode')
```

Comment:
Add explanation why this is happening. Plus, we can probably just do it in one line